src/nft/components/NFTLifecycleProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import { createContext, useContext, useEffect } from 'react';
+import { useValue } from '../../internal/hooks/useValue';
+import { useLifecycleStatus } from '../hooks/useLifecycleStatus';
+import type {
+  NFTLifecycleContextType,
+  NFTLifecycleProviderReact,
+} from '../types';
+
+const emptyContext = {} as NFTLifecycleContextType;
+
+export const NFTLifecycleContext =
+  createContext<NFTLifecycleContextType>(emptyContext);
+
+export function useNFTLifecycleContext() {
+  const context = useContext(NFTLifecycleContext);
+  if (context === emptyContext) {
+    throw new Error(
+      'useNFTLifecycleContext must be used within an NFTView or NFTMint component',
+    );
+  }
+  return context;
+}
+
+export function NFTLifecycleProvider({
```

Comment:
i like how you abstracted this away. i think we should follow the same pattern in our other components.

Reply:
+1, this is great